docs-compose/overwatch_docs/content/DeployOverwatch/PipelineValidationFramework/DeepDiveValidationFramework.md

```diff
@@ -0,0 +1,194 @@
+* Job_Run_Cluster_Util_value_Should_Not_Be_More_Than_One
+* Check_Whether_Any_Job_is_Running_For_Multiple_Days
+
+#### cluster_gold
+* Cluster_ID_Should_Not_be_NULL
+* Driver_Node_Type_ID_Should_Not_be_NULL
+* Node_Type_ID_Should_Not_be_NULL_for_Multi_Node_Cluster
+* Cluster_Type_Should_be_In_Between_Serverless_SQL-Analytics_Single-Node_Standard_High-Concurrency
+* 
+
+#### sparkjob_gold
+* Cluster_ID_Should_Not_be_NULL
+* Job_ID_Should_Not_be_NULL
+* db_id_in_job_Should_Not_be_NULL_When_db_Job_Id_is_Not_NULL
+
+#### sql_query_history_gold
+* Warehouse_ID_Should_Not_be_NULL
+* Query_ID_Should_Not_be_NULL
+
+#### jobrun_gold
+* Job_ID_Should_Not_be_NULL
+* Run_ID_Should_Not_be_NULL
+* Job_Run_ID_Should_Not_be_NULL
+* Task_Run_ID_Should_Not_be_NULL
+* Cluster_ID_Should_Not_be_NULL
+
+#### job_gold
+* Job_ID_Should_Not_be_NULL
+* Action_Should_be_In_Between_snapimpute_create_reset_update_delete_resetJobAcl_changeJobAcl
+
+### Cross_Table_Validation
+The healthCheck_rules with this rule_type would be applied to 2 or more tables. This rule type is basically used to 
+validated data consistency across tables.
+
+HealthCheck_rules of this rule_type are as below:
+* JOB_ID_Present_In_jobRun_gold_But_Not_In_jobRunCostPotentialFact_gold
+* JOB_ID_Present_In_jobRunCostPotentialFact_gold_But_Not_In_jobRun_gold
+* CLUSTER_ID_Present_In_jobRun_gold_But_Not_In_jobRunCostPotentialFact_gold
+* CLUSTER_ID_Present_In_jobRunCostPotentialFact_gold_But_Not_In_jobRun_gold
+* NOTEBOOK_ID_Present_In_notebook_gold_But_Not_In_notebookCommands_gold
+* NOTEBOOK_ID_Present_In_notebookCommands_gold_But_Not_In_notebook_gold
+* CLUSTER_ID_Present_In_clusterStateFact_gold_But_Not_In_jobRunCostPotentialFact_gold
+* CLUSTER_ID_Present_In_jobRunCostPotentialFact_gold_But_Not_In_clusterStateFact_gold
+* CLUSTER_ID_Present_In_cluster_gold_But_Not_In_clusterStateFact_gold
+* CLUSTER_ID_Present_In_clusterStateFact_gold_But_Not_In_cluster_gold
+
+
+## How to Run Validation Framework
+Currently, we can execute validation framework through notebook only. Depend on requirement running through databricks workflow
+would be added in the future.
+
+Execution would be started upon calling `PipelineValidation()` function. Below are the input arguments of this function:
+| Param                | Type    | Optional | Default Value | Description                                                                 |
+|----------------------|---------|----------|---------------|-----------------------------------------------------------------------------|
+| etlDB                | String  | No       | NA            | Overwatch etl database name on which Validation Framework need to be run    |
+| tableArray           | String  | Yes      | Array()       | Array of tables on which Single Table Validation need to be performed       |
+| crossTableValidation | Boolean | Yes      | Yes           | Boolean flag act as on and off switch for cross table validation            |
+
```

Comment:
for the tableArray i guess the default value should be all the tables which is eligible for Validation.

Reply:
@sriram251-code here by default we are taking all tables eligible for validation. Only thing is here rather than mentioning all the tables I am putting it as empty so that when we get the length of array as 0 validation would run on all tables.

Reply:
please add it in description column.

Reply:
@sriram251-code changes are done.